### ISSUE TITLE
Tilt-to-wake not working

### ISSUE THREAD

**Issue by jes** [AsteroidOS/meta-smelt-hybris#9]
It's quite likely that I simply don't know how to set it up, but no matter what I do, I can't seem to see the watchface just by tilting the watch?

AsteroidOS is great btw. I found the install process very straightforward, after I'd managed to pry the crusted-up glue off the back of this watch :)

**Comment by MagneFire** [comment]
Huh, that's weird... Can you confirm that you have the Tilt-to-Wake setting in the settings app?

If so. Can you check that /system/bin/logcat shows an gesture event?

It may also be possible that you have to wait a while for the sensor firmware to be uploaded, assuming that you have Android Wear Nougat installed. Maybe wait around 10 minutes?

**Comment by jes** [comment]
Hi, I can't find the Tilt-To-Wake setting. I have the following Settings:

 - Time
 - Data
 - Language
 - Bluetooth
 - Brightness
 - Units
 - Wallpaper
 - Watchface
 - USB
 - Power Off
 - Reboot
 - About

I don't think I'll be able to run logcat because I can't pair it with my phone and therefore can't turn on Wi-Fi.

**Comment by jes** [comment]
Update: I have paired it with my phone but still don't think I understand how to turn no Wi-Fi.

**Comment by MagneFire** [comment]
You should be able to connect to the watch via SSH: https://asteroidos.org/wiki/ssh/ then just execute the logcat command. Or provide a `dmesg`. Have you done a real installation or a temporary one. If you have a real install, can you reboot the watch to see if that changes things?

As for WiFi this is something that is not really used with AsteroidOS. I mean you can enable it if you have access to it via SSH. But it does not really give you any benefit. It is more something that is reserved for future use :)

**Comment by jes** [comment]
It's a full install, although it's super inconvenient to use USB on this watch because I'd need to peel the back off again and solder the wires back on. I'd rather use Wi-Fi.

That said, rebooting the watch did make Tilt-to-wake start working! Thanks for the tip. :)

In the absence of any other info, I think I will proceed by trying to connect it over USB & configure Wi-Fi, and then I should be able to write & install my own apps using SSH over Wi-Fi, right?

What I want to be able to do is write my own apps for the watch and install them without connecting to USB, is that something I'm likely to be able to do with AsteroidOS, or is the technology not yet at that point? I wonder how convenient it would be to install a permanent connector to make it a bit more convenient to connect to USB...

**Comment by MagneFire** [comment]
Great to so Tilt-to-Wake working after all!

I know USB is kind of a sore point on this watch. That's why I use a 3D printed dock 😊

You should indeed be able to install apps using scp and ssh over WiFi although I am not sure how stable WiFi will be for full time use (deep sleep issues may arise). USB is the preferred way...

As for installing apps on the watch. That is definitely possible. You may try to adjust existing apps or build sdl2 apps. This should you get started: https://asteroidos.org/wiki/creating-an-asteroid-app/

The procedure will end up to something like:
- Setup build environment
- Create custom app
- Build app using bitbake
- Use scp to send ipk to watch
- Use ssh to install the ipk
- Start app using newly added shortcut on the app list

**Comment by jes** [comment]
Thanks, I'll attempt working through that. FWIW, I couldn't get Wi-Fi to work. The "wpa_cli" just reports "FAIL" whenever I try to "scan". I expect I need to do the part about building a new image, which seems beyond my skillset.

In trying to build the app development environment, I got stuck at the "bitbake meta-toolchain-qt5" step for a while, and found this thread: https://community.nxp.com/thread/532638 - basically the "stime()" function has been removed in the glibc version available on Ubuntu 20.04, but the fix mentioned in that thread worked. ~/asteroid/build/tmp-glibc/work/x86_64-linux/qemu-native/3.1.1.1-r0/qemu-3.1.1.1/linux-user/syscall.c line 7404 now looks like:

`            return get_errno(clock_settime(CLOCK_REALTIME,&host_time));`

and the build seems to be getting further.
